From c447fc22037c1a00ebb7057290a4cf9772953f14 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mika=20Bostr=C3=B6m?= <mika.bostrom@nomovok.com>
Date: Thu, 19 Apr 2012 18:42:04 +0300
Subject: [PATCH 2/6] Fix build in dedicated build root

Try to remove abuses of DESTDIR macro and provide a way to cleanly
include headers from 3D submodules without forcing new files to other
build targets.
---
 src/imports/threed/threed.pro              |    4 +++-
 src/plugins/sceneformats/assimp/assimp.pro |    4 ++--
 src/plugins/sceneformats/bezier/bezier.pro |    3 ++-
 src/plugins/sceneformats/obj/obj.pro       |    3 ++-
 src/quick3d/quick3d.pro                    |    2 ++
 src/threed/includes.pri                    |   15 +++++++++++++++
 src/threed/threed.pro                      |    1 +
 7 files changed, 27 insertions(+), 5 deletions(-)
 create mode 100644 src/threed/includes.pri

diff --git a/src/imports/threed/threed.pro b/src/imports/threed/threed.pro
index 06a2ffb..513a66f 100644
--- a/src/imports/threed/threed.pro
+++ b/src/imports/threed/threed.pro
@@ -4,7 +4,9 @@ CONFIG += qt plugin
 
 QT += quick qml 3d 3dquick
 
-DESTDIR = $$QT.3dquick.imports/Qt3D
+include(../../threed/includes.pri)
+
+#DESTDIR = $$QT.3dquick.imports/Qt3D
 target.path = $$[QT_INSTALL_IMPORTS]/Qt3D
 INSTALLS += target
 
diff --git a/src/plugins/sceneformats/assimp/assimp.pro b/src/plugins/sceneformats/assimp/assimp.pro
index 5a0fca5..572b220 100644
--- a/src/plugins/sceneformats/assimp/assimp.pro
+++ b/src/plugins/sceneformats/assimp/assimp.pro
@@ -4,6 +4,7 @@ QT += 3d
 load(qt_plugin)
 
 include(../../qpluginbase.pri)
+include(../../../threed/includes.pri)
 HEADERS += qailoader.h \
     qaiscene.h \
     qaiscenehandler.h \
@@ -24,8 +25,7 @@ OTHER_FILES += \
     ai_scene_plugin.json
 
 
-
-DESTDIR = $$QT.3d.plugins/sceneformats
+#DESTDIR = $$QT.3d.plugins/sceneformats
 target.path = $$[QT_INSTALL_PLUGINS]/sceneformats
 INSTALLS += target
 
diff --git a/src/plugins/sceneformats/bezier/bezier.pro b/src/plugins/sceneformats/bezier/bezier.pro
index 870d697..18525fe 100644
--- a/src/plugins/sceneformats/bezier/bezier.pro
+++ b/src/plugins/sceneformats/bezier/bezier.pro
@@ -4,6 +4,7 @@ QT += 3d
 load(qt_plugin)
 
 include(../../qpluginbase.pri)
+include(../../../threed/includes.pri)
 
 HEADERS += qglbezierscene.h \
            qglbezierscenehandler.h \
@@ -16,6 +17,6 @@ SOURCES += \
 OTHER_FILES += \
             bezier_scene_plugin.json
 
-DESTDIR = $$QT.3d.plugins/sceneformats
+#DESTDIR = $$QT.3d.plugins/sceneformats
 target.path = $$[QT_INSTALL_PLUGINS]/sceneformats
 INSTALLS += target
diff --git a/src/plugins/sceneformats/obj/obj.pro b/src/plugins/sceneformats/obj/obj.pro
index 73bd0e8..c48fe07 100644
--- a/src/plugins/sceneformats/obj/obj.pro
+++ b/src/plugins/sceneformats/obj/obj.pro
@@ -4,6 +4,7 @@ QT += 3d
 load(qt_plugin)
 
 include(../../qpluginbase.pri)
+include(../../../threed/includes.pri)
 
 HEADERS += qglobjscene.h \
            qglobjscenehandler.h \
@@ -16,6 +17,6 @@ SOURCES += \
 OTHER_FILES += \
            obj_scene_plugin.json
 
-DESTDIR = $$QT.3d.plugins/sceneformats
+#DESTDIR = $$QT.3d.plugins/sceneformats
 target.path = $$[QT_INSTALL_PLUGINS]/sceneformats
 INSTALLS += target
diff --git a/src/quick3d/quick3d.pro b/src/quick3d/quick3d.pro
index e3d4de8..f4026e6 100644
--- a/src/quick3d/quick3d.pro
+++ b/src/quick3d/quick3d.pro
@@ -28,6 +28,8 @@ qt3d_qml_deploy_dir {
 }
 
 include(quick3d.pri)
+include(../threed/includes.pri)
+LIBS += -L../threed/ -lQt3D
 
 PUBLIC_HEADERS = $$HEADERS
 HEADERS += $$PRIVATE_HEADERS
diff --git a/src/threed/includes.pri b/src/threed/includes.pri
new file mode 100644
index 0000000..5c8cabe
--- /dev/null
+++ b/src/threed/includes.pri
@@ -0,0 +1,15 @@
+# Add everything in threed/* to include paths
+INCLUDEPATH += $$PWD/../quick3d
+INCLUDEPATH += $$PWD/api
+INCLUDEPATH += $$PWD/arrays
+INCLUDEPATH += $$PWD/effects
+INCLUDEPATH += $$PWD/geometry
+INCLUDEPATH += $$PWD/global
+INCLUDEPATH += $$PWD/graphicsview
+INCLUDEPATH += $$PWD/materials
+INCLUDEPATH += $$PWD/math3d
+INCLUDEPATH += $$PWD/painting
+INCLUDEPATH += $$PWD/scene
+INCLUDEPATH += $$PWD/surfaces
+INCLUDEPATH += $$PWD/textures
+INCLUDEPATH += $$PWD/viewing
diff --git a/src/threed/threed.pro b/src/threed/threed.pro
index e6eb3a6..7f82777 100644
--- a/src/threed/threed.pro
+++ b/src/threed/threed.pro
@@ -19,6 +19,7 @@ gcov {
 
 include(../private/private.pri)
 include(threed.pri)
+include(includes.pri)
 PUBLIC_HEADERS = $$HEADERS
 HEADERS += $$PRIVATE_HEADERS
 DEFINES += QT_BUILD_QT3D_LIB
-- 
1.7.5.4

